mne/viz/_3d.py

```diff
@@ -1344,3 +1345,135 @@ def _get_view_to_display_matrix(scene):
                                  [0.,            0.,   1.,        0.],
                                  [0.,            0.,   0.,        1.]])
     return view_to_disp_mat
+
+
+def plot_dipole_3d(dipole, trans, subject, subjects_dir=None, scale_factor=1e9,
+                   block=False, show=True):
+    """Plot dipoles in 3-D. Browse through the dipoles using mouse scroll or
+    up/down arrows.
+
+    Parameters
+    ----------
+    dipole : instance of mne.Dipole
+        The dipole to plot.
+    trans : dict
+        The mri to head trans.
+    subject : str
+        The subject name corresponding to FreeSurfer environment variable
+        SUBJECT.
+    subjects_dir : None | str
+        The path to the freesurfer subjects reconstructions. It corresponds to
+        Freesurfer environment variable SUBJECTS_DIR. The default is None.
+    scale_factor : float
+        The scaling applied to convert amplitudes to voxel sizes for the plot.
+        Defaults to 1e9.
+    block : bool
+        Whether to halt program execution until the figure is closed. Defaults
+        to False.
+    show : bool
+        Show figure if True. Defaults to True.
+
+    Returns
+    -------
+    fig : instance of matplotlib figure
+        The figure containing 3-D locations of the dipoles.
+    """
+    import matplotlib.pyplot as plt
+    from mpl_toolkits.mplot3d import Axes3D  # noqa
+    try:
+        import nibabel as nib
+    except ImportError:
+        raise
+    subjects_dir = get_subjects_dir(subjects_dir=subjects_dir,
+                                    raise_error=True)
+    t1_fname = op.join(subjects_dir, subject, 'mri', 'T1.mgz')
+    t1 = nib.load(t1_fname)
+    ras2vox = t1.header.get_ras2vox()
+    data = t1.get_data()
+    dims = len(data)  # Symmetric size assumed.
+    gridx, gridy = np.meshgrid(np.linspace(0, dims, dims),
+                               np.linspace(0, dims, dims))
+
+    fig = plt.figure()
+    ax = fig.add_subplot(111, projection='3d')
+
+    trans = _get_trans(trans, fro='head', to='mri')[0]
```

Comment:
If possible there should be a `coord_frame` argument. It's might actually be most natural to plot these in `head` coords